docs/src/references/operation-workflow.md

```diff
@@ -0,0 +1,376 @@
+    so, a domain specific component can be substituted to add extra checks and actions before moving forward. 
+- Extra states and participants can be added as long as each state is owned by one participant.
+  - For instance, an agent developer can introduce a `Rollback` state in the `configuration-update` workflow,
+    associated by another software component responsible for these rollbacks.
+
+Furthermore, specific versions of the same workflow can be defined on different targets.
+Indeed, all the status updates for a command on a given thin-edge entity or component
+are published on an MQTT topic which prefix is the entity identifier.
+- The same executable can be used to handle operations on different targets.
+  For instance, the `tedge-configuration-plugin` can run on the main device `te/device/main//`
+  as well as on a child-device identified by `te/device/child-xyz//`.
+- A specific executable can be substituted on a specific target.
+  If for some reasons, `tedge-configuration-plugin` cannot be installed on a child-device,
+  then a specific implementation of the `configuration-update` MQTT API can be used to serve configuration updates
+  on, say, `te/micro-controller/xyz//`.
+- A workflow can be extended differently for each target.
+  As an example, an agent developer can define an extra rollback state on the main device but not on the child devices.
+
+### Operation API
+
+As several software components have to collaborate when executing a command, each operation must define a specific API.
+This API should be based on the principles of MQTT-driven workflow and defines: 
+- the well-known operation name such `restart` or `software-update`
+- user documentation of the required input and the expected outcome of an operation request
+- the set of observable states for a command and the possible state sequences
+- for each state:
+  - the well-known name such as `Download` or `Downloaded`
+  - the schema of the state payload and the required parameters to process a command at this stage 
+  - developer documentation on the role of each parameter and the expected checks and actions
+- the schema for the capability message sent when the operation is enabled on some thin-edge entity or component
+  - developer documentation on the role of each field of the capability message
+
+A workflow implementation is free to define the states a command can go through
+as well as the message payload attached to each state.
+
+However, there are some rules and best practices.
+
+- Three states are mandatory: `Init`, `Successful` and `Failed`.
+- `Init` must be the unique initial state of the workflow.
+  - This state must give all the information required to execute the command.
+  - Having no other initial states is important so any command requester (as the mapper)
+    can trigger commands in a systematic manner *even* if the workflow is updated.
+- `Successful` and `Failed` must be the unique terminal states of the workflow.
```

Comment:
Indeed something needs to be clarified.

By `failed` terminal step, I mean a state for the final conclusion returned to the requester : its command failed. If some rollback actions have been performed, these must have been done *before*. In practice, this means that the workflow designer will have to introduce specific states to handle rollbacks. Some extra states along these lines: `executing` -> `rollback-required` -> `rollbacking` -> `failed`.

It's important to have well-known fixed terminal states, otherwise the mapper will have no way to know when the execution is actually finished.